modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/defragmentation/TimeTracker.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.persistence.defragmentation;
+
+import java.util.Arrays;
+import org.apache.ignite.internal.util.typedef.internal.U;
+
+/** */
+public class TimeTracker<Stage extends Enum<Stage>> {
```

Comment:
Is this the same as `TimeBag`?

Reply:
It kinda is, but it's not. Here we accumulate time of the same stages over many iterations. For example, we can have 1000 checkpoint read lock acquirings and 100000 cache data entries readings. Technically, each of those are a separate "stage", but what we need is only the sum of timings.

Reply:
Can you clarify why we need accumulated timings per each stage?

Reply:
I mean that I need sum of all times that we waited for lock to be acquired, not every single time value.

Reply:
Oh, I got it wrong again. We need it for debugging and finding hot spots.

Reply:
I think we shouldn't have classes with 'almost the same behaviour'. 

1. AFAIK, the `TimeBag` collects all the results per each Stage and then aggregates it. If you write a custom aggregate function will you get the same result as the `TimeTracker` does?

1. Why do we need any aggregated results? Can we dump raw timings to log and aggregate than by some external tool (log-parser)? I think raw results would provide much valuable debug information

Reply:
1. How can I use custom aggregating function for TimeBag? I don't see anything like that in the code. It all is very convoluted, TimeBag is designed for processes that have no repeating stages.
2. I don't want to have 100 thousand entries in log to aggregate them later, it's too much. And current class aggregates time as effective as it's possible, pretty much. It doesn't use lists and maps and all these heave things.

Reply:
Thas why any build-in aggregation machinery is a bad idea. For building reports, the raw data must be used.

I have no doubts about the necessity of having things for debugging and measuring performance, finding hot spots but let's use more flexible and standard things like - [Tracing](https://issues.apache.org/jira/browse/IGNITE-13060) and/or [Cluster performance profiling tool](https://issues.apache.org/jira/browse/IGNITE-12666)

Let's remove this thing from PR and file a new issue to implement it the right and more flexible way.

Reply:
Ok, I still don't get why you want this code being removed so badly. I'm removing it